Documentation/Manual/Texinfo/extensions.txi

```diff
@@ -222,3 +223,171 @@ image. Valid values are @code{:png}, @code{:jpg}, @code{:jpeg},
+200 units left margin will be located at 67th unit.
+
+Sometimes stream is associated with a sheet which doesn't have a
+bounding rectangle (for instance @code{:eps} stream on PostScript
+backend has sheet's region @code{+everywhere}). In such case margins
+are relative to these arbitrary positions:
+
+@itemize
+@item left margin --- x = 0
+@item top margin --- y = 0
+@item right margin --- x = 80 * default-text-style-width
+@item bottom margin --- y = 43 * default-text-style-height
+@end itemize
+@end deffn
+
+@deffn {Function} {clime:stream-effective-left-margin} stream
+@deffnx {Function} {clime:stream-effective-top-margin} stream
+@deffnx {Function} {clime:stream-effective-right-margin} stream
+@deffnx {Function} {clime:stream-effective-bottom-margin} stream
+To probe the stream margin position in its native coordinates we call
+a function @code{stream-effective-*-margin}. These functions always
+return a number.
+@end deffn
+
+@deffn {Macro} {clime:with-temporary-page} (stream margin-spec &key move-cursor) &body body
+Function temporarily auguments stream margins according to
+@code{margin-spec} (margins which are not specified are inherited from
+the current stream configuration). Macro changes cursor position to
+the @code{page-cursor-initial-position}. If @code{move-cursor} is
+@code{T} (default) cursor is left where it was at the end of the macro
+scope. Otherwise old stream-cursor-position is reinstationed.
+@end deffn
+
+@subsection Line and page wrapping
+
+Word wrapping in CLIM 2.0 can be achieved with @code{filling-output}
+macro. McCLIM in addition to that supports @code{:end-of-line-action}
+called @code{:wrap*}. It breaks lines by a word when the string width
+exceeds line length between left and right margins.
+
+Both @code{indenting-output} and @code{filling-output} are implemented
+in terms of the @code{with-temporary-page} macro. Former temporarily
+auguments the left margin while the latter does the same for the right
+one. @code{filling-output} has a behavior extending what is defined in
+CLIM 2.0.
+
+@deffn {Macro} {clim:filling-output} stream &key fill-width breah-characters after-line-break after-line-break-initially after-line-break-subsequent
+
+Macro behaves as specified in CLIM 2.0 specification. Additionally
+@code{after-line-break} may be a function accepting two arguments:
+stream and a boolean indicating whenever line was broken by a hard
+break or a soft newline due to word wrap. String is printed (or
+function is executed) with the same ink and text-style as was set on
+the stream when @code{filling-output} scope has started.
+
+If @code{after-line-break-initially} is @code{T}
+@code{after-line-break} is triggered on each ``hard''
+newline. Otherwise it is not.
+
+If @code{after-line-break-subsequent} is @code{T} (default)
+@code{after-line-break} is triggered on each ``soft''
+newline. Otherwise it is not.
+@end deffn
+
+In the following example we print all lines in @code{*lorem-ipsum*}
```

Comment:
You probably mean `clim-demo::*lorem-ipsum*`. It would be better to define the variable in the documentation itself, so it doesn't depend on `clim-demo` being loaded.

Reply:
I didn't refer any particular variable, it is just an illustration code. you are right though, making it "working" out of the box makes sense.